frontend/app/services/SubscriptionService.scala

```diff
@@ -102,79 +105,96 @@ object SubscriptionService {
   }
 }
 
-class SubscriptionService(val tierPlanRateIds: Map[ProductRatePlan, String],
-                          val zuoraSoapService: ZuoraSoapService,
+class SubscriptionService(val zuoraSoapService: ZuoraSoapService,
                           val zuoraRestService: ZuoraRestService) extends AmendSubscription with LazyLogging {
-  import SubscriptionService._
-
-  def getAccount(memberId: MemberId): Future[Account] =
-    zuoraSoapService.query[Account](s"crmId='${memberId.salesforceAccountId}'").map(sortAccounts(_).last)
-
-  //TODO before we do subs we need to filter by rate plans membership knows about
-  def getSubscriptions(memberId: MemberId): Future[Seq[Subscription]] = {
-    for {
-      account <- getAccount(memberId)
-      subscriptions <- zuoraSoapService.query[Subscription](s"AccountId='${account.id}'")
-    } yield subscriptions
-  }
-
-  def memberTierFeatures(memberId: MemberId): Future[Seq[Rest.Feature]] =
-    for {
-      account <- getAccount(memberId)
-      features <- zuoraRestService.productFeaturesByAccount(account.id)
-    } yield features
-
-  def getSubscription(subscriptionId: String): Future[Subscription] = zuoraSoapService.queryOne[Subscription](s"Id='$subscriptionId'")
-
-  def getSubscriptionStatus(memberId: MemberId): Future[SubscriptionStatus] = {
-    for {
-      subscriptions <- getSubscriptions(memberId)
 
-      if subscriptions.nonEmpty
+  import SubscriptionService._
+  val membershipProductType = "Membership"
+  val productRatePlanChargeModel = "FlatFee"
+
+  val tierPlanRateIds: ProductRatePlan => Future[String] = productRatePlan =>
```

Comment:
Can we call this `tierRatePlanId`? it returns a single id, not a collection thereof. Also, is there any benefit in creating an anonymous function here instead of just a method using `def`?